### DISCUSSION TITLE
support content-dispostion response header using s3 api

### DISCUSSION THREAD

**Discussion by pkk-code** [seaweedfs/seaweedfs#2502] [Q&A] [answered]
I'm trying to set content-dispostion response header on get using s3 presigned url api with seaweedfs endpoint.  

const presignedDownload = () => {
  const docName = "myownname.txt"
  return s3.getSignedUrl("getObject", {
    Bucket: "testbucket",
    Key: "/3456/61aa6fefbb6dc7316017a485",
    Expires: 100, 
    ResponseContentDisposition: `attachment; filename="${docName}"`
  });
};

The response is  inline; Content-Disposition filename="61aa6fefbb6dc7316017a485". Expected behaviour filename to be myownname.txt

I don't see this is set correctly in response headers. Can this be supported to set the correct Content-Disposition. Thanks

**Comment by chrislusf** [comment, marked as answer]
Thanks for reporting! I added a fix. https://github.com/chrislusf/seaweedfs/commit/7bf14452f7882c803c7c86dcb7b81be65bf2934e
But not sure whether the fix works.
Please have a working code so that I can reproduce this.

**Reply by pkk-code** [reply]
Thanks for getting back to me. It is still not setting the right content-disposition header.  Here is the sample nodejs code to reproduce this.

//Sample
const AWS = require("aws-sdk");
var ep = new AWS.Endpoint("http://localhost:8333"); //seaweedfs endpoint
var s3 = new AWS.S3({
  accessKeyId: "", //access key
  secretAccessKey: "", //secret
  endpoint: ep,
  s3ForcePathStyle: true,
  region: "us-east-1",
  signatureVersion: "v4",
});
const params = {
  Bucket: "testnodejs", //provide bucket name
  Key: "udm_seaweedfs/61aa/61aa6fefbb6dc7316017a485", //provide key
  Expires: 100,
  ResponseContentDisposition: `attachment; filename="myownname.txt"`,
};
//   getSingedUrl()
const presignedDownload = () => {
  return s3.getSignedUrl("getObject", params);
};
console.log(presignedDownload()); // Call function to generate pre-signed url.
// node download.js

![image](https://user-images.githubusercontent.com/37810104/145843638-8cadce72-ffff-4d26-98b7-5ad259ed278e.png)

**Reply by chrislusf** [reply]
How to run the js code?

**Reply by pkk-code** [reply]
Please save this code to a file like download.js and run it using **node download.js**. 
You may also need to execute this statement npm install aws-sdk to have the dependency installed.

**Reply by pkk-code** [reply]
[download.go.zip](https://github.com/chrislusf/seaweedfs/files/7707455/download.go.zip)

Incase if you want to reproduce with go. Thanks.

**Reply by chrislusf** [reply]
You go example code did not send any requests to SeaweedFS S3 at all.

**Reply by pkk-code** [reply]
Sorry, i had an environment variable on my system to pass the seaweedfs endpoint. 
Please try with this now. This points now to 	awsEndpoint = "http://localhost:8333/" 
and  pass the credentials if needed 
creds := credentials.NewStaticCredentialsProvider("", "", "") 


[download.go.zip](https://github.com/chrislusf/seaweedfs/files/7719878/download.go.zip)

**Reply by chrislusf** [reply]
I have changed it already. Actually you can set it to anything, it still works.

**Reply by pkk-code** [reply]
OK. You still see its not routed to seaweedfs? 
I'm lost how presigning is happening and content of file stored in seeweedfs is accessible with generated url.  Is there a way, i can generate the log from seaweedfs s3 on my end and pass it to you for further debugging?

**Reply by chrislusf** [reply]
No requests received.

You can see existing logs with "weed -v=4 s3 ..."

**Reply by pkk-code** [reply]
When i access the generated presigned url i see this in seaweed log. 

I1215 12:58:13 95418 auth_credentials.go:216] v4 auth type
I1215 12:58:13 95418 auth_credentials.go:237] user name: admin actions: [Admin Read List Tagging Write], action: Read
I1215 12:58:13 95418 s3api_object_handlers.go:136] GetObjectHandler testnodejs /udm_seaweedfs/61aa/61aa6fefbb6dc7316017a485
I1215 12:58:13 95418 s3api_object_handlers.go:320] s3 proxying GET to http://192.168.86.134:8888/buckets/testnodejs/udm_seaweedfs/61aa/61aa6fefbb6dc7316017a485

**Reply by chrislusf** [reply]
added a fix https://github.com/chrislusf/seaweedfs/commit/7210558c7bb6efbd8377975918143478a14c5b3c

**Reply by pkk-code** [reply]
It worked! 

Appreciate your help in fixing this. Awesome.
